packages/network-controller/src/NetworkController.ts

```diff
@@ -1965,12 +1992,15 @@ export class NetworkController extends BaseController<
           EIPS: {},
         };
       }
-      const meta = state.networksMetadata[networkClientId];
-      meta.status = networkStatus;
-      if (isEIP1559Compatible === undefined) {
-        delete meta.EIPS[1559];
-      } else {
-        meta.EIPS[1559] = isEIP1559Compatible;
+      const newMetadata = state.networksMetadata[networkClientId];
+      newMetadata.status = metadata.networkStatus;
+
+      if ('isEIP1559Compatible' in metadata) {
```

Comment:
Is it intended that we no longer reset `isEIP1559Compatible` if it's omitted from the metadata object? That is a functional change from before

Reply:
Oh I see, it is intended, I get it now. The old invocations all include this property, so this doesn't functionally change those invocations.